src/tree/updater_gpu_hist.cu

```diff
@@ -568,11 +621,32 @@ struct GPUHistMakerDevice {
     auto right_weight = node_value_constraints[candidate.nid].CalcWeight(
                             param, candidate.split.right_sum) *
                         param.learning_rate;
-    tree.ExpandNode(candidate.nid, candidate.split.findex,
-                    candidate.split.fvalue, candidate.split.dir == kLeftDir,
-                    base_weight, left_weight, right_weight,
-                    candidate.split.loss_chg, parent_sum.GetHess(),
-                     candidate.split.left_sum.GetHess(), candidate.split.right_sum.GetHess());
+
+    auto is_cat = candidate.split.is_cat;
+    if (is_cat) {
+      auto cat = common::AsCat(candidate.split.fvalue);
```

Comment:
Note to myself: in one-hot encoded setting, there is only one matching category in every categorical split. However, the `split_categories_` structure can later store multiple matching categories per split.

Reply:
Another reminder to myself: Treelite must support JSON format of XGBoost.